app/services/compute_resource_cache.rb

```diff
@@ -0,0 +1,89 @@
+# This class caches attributes for a compute resource in
+# rails cache to speed up slow or expensive API calls
+class ComputeResourceCache
+  attr_accessor :compute_resource, :cache_duration
+
+  delegate :logger, :to => ::Rails
+
+  def initialize(compute_resource, cache_duration: nil)
+    self.compute_resource = compute_resource
+    self.cache_duration = cache_duration || 180.minutes
```

Comment:
Instead of defaulting the kwarg to nil then defaulting here to 180.minutes, you can do that in the kwarg itself